support-frontend/assets/helpers/abTests/abtest.ts

```diff
@@ -251,44 +366,45 @@ const trackOptimizeExperiment = (
 	);
 };
 
+type VariantAssignment =
+	| { type: 'NOT_ASSIGNED' }
+	| { type: 'ASSIGNED'; variantIndex: number };
+
 function assignUserToVariant(
 	mvtId: number,
 	test: Test,
-	testId: string,
-	acquisitionDataTests: AcquisitionABTest[] | undefined,
-): number {
-	const { referrerControlled, seed } = test;
+	participation: ActiveParticipation,
+): VariantAssignment {
+	// For non-referrrer controlled tests we assign the user randomly
+	if (participation.type === 'ACTIVE_PARTICIPATION') {
+		return assigned(randomNumber(mvtId, test.seed) % test.variants.length);
+	}
 
-	if (referrerControlled && acquisitionDataTests != null) {
-		const acquisitionDataTest = acquisitionDataTests.find((t) =>
-			t.name.startsWith(testId),
-		);
+	// For referrrer controlled tests the assignment comes from the acquisition data.
+	// If the variant in the acquisition data doesn't match up with any defined in
+	// the test, we log an error and don't assign the user to the test.
+	const acquisitionVariant = participation.acquisitionDataTest.variant;
 
-		if (!acquisitionDataTest) {
-			return -1;
-		}
```

Comment:
This is the check we had to do before, but is actually redundant as we know if the user is participating in this test that it must be in the array of tests. Using this new type we no longer need to do this.